File: 00_course_material/assignments/01a_data_parsing_server_part_1/Node_solution/dataParser.js
const Papa = require('papaparse'); //bruger is ESmodules men commonJS. I fremtiden brug kun ESmodules
const fileSystem = require('fs');
const path = require('path');
const yaml = require('js-yaml');
const csv = require('fast-csv');
const xml = require('xml2js');

const folderPath = "../Data"

async function handleFiles(folderPath) {
    try {
        const files = fileSystem.readdirSync(folderPath);

        for(const file of files ){
            const filePath = path.join(folderPath, file);
            const fileExtension = path.extname(file);

            const parsedData = await parseFile(filePath, fileExtension);
            console.log(`Parsed Data for ${file}:\n`, parsedData);
        }
    } catch (error) {
        console.error(error);
    }
}

async function parseFile(filePath, fileExtension) {
    try {
        const fileContent = fileSystem.readFileSync(filePath, 'utf-8');

        switch(fileExtension) {
            case '.yaml':
            case '.yml':
                return yaml.load(fileContent);
            case '.csv':
                const parsedCsv = Papa.parse(fileContent);
                return parsedCsv.data;
            case '.xml':
                return xml.parseStringPromise(fileContent);
            case '.json':
                return JSON.parse(fileContent);
            case '.txt':
                return fileContent;
            default:
                throw new Error('Unsupported file extension');
        }
    } catch (error) {
        console.error(error);
    }
}

handleFiles(folderPath);

File: 00_course_material/assignments/01a_data_parsing_server_part_1/Python_solution/dataParser.py
import os
import json
import csv
import xml.etree.ElementTree as ET
import yaml
import io

path = '00_course_material/assignments/01a_data_parsing_server_part_1/Data'

def read_files(path):
    try:
        files = os.listdir(path)
        for file in files:
            file_path = os.path.join(path, file)
            file_extension = file.split('.')[-1]
            parsed_data = parse_files(file_path, file_extension)
            print(parsed_data)
    except Exception as e:
        print(e)
        return None

def parse_files(file_path, extension):
    try:

        with open(file_path, 'r', encoding='utf-8') as f:
            fileContent = f.read()
    
        if extension == 'json':
            return json.loads(fileContent)
        elif extension == 'csv':
            return list(csv.reader(io.StringIO(fileContent)))
        elif extension == 'xml':
            return ET.fromstring(fileContent)
        elif extension in['yaml','yml']:
            return yaml.safe_load(fileContent)
        elif extension == 'txt':
            return fileContent
        else :
            return print('Invalid extension')
    except Exception as e:
        print(e)
        return None
    
read_files(path)